test/alternator/test_streams.py

```diff
@@ -735,17 +781,95 @@ def do_test(test_table_ss_stream, dynamodbstreams, updatefunc, mode, p = random_
-@pytest.mark.xfail(reason="Currently fails - see issue #6930")
+# event. Reproduces #6930.
 def test_streams_putitem_keys_only(test_table_ss_keys_only, dynamodbstreams):
     def do_updates(table, p, c):
         events = []
         table.put_item(Item={'p': p, 'c': c, 'x': 2})
-        events.append(['INSERT', {'p': p, 'c': c}, None, {'p': p, 'c': c, 'x': 2}])
+        # TODO: change to 'INSERT' when #6918 is fixed
+        events.append([['INSERT', 'MODIFY'], {'p': p, 'c': c}, None, {'p': p, 'c': c, 'x': 2}])
         return events
     do_test(test_table_ss_keys_only, dynamodbstreams, do_updates, 'KEYS_ONLY')
 
+def test_streams_putitem_new_item_overrides_old_lsi(test_table_sss_new_and_old_images_lsi, dynamodbstreams):
+    def do_updates(table, p, c):
+        events = []
+        table.put_item(Item={'p': p, 'c': c, 'a': '1'})
+        # TODO: change to 'INSERT' when #6918 is fixed
+        events.append([['INSERT', 'MODIFY'], {'p': p, 'c': c}, None, {'p': p, 'c': c, 'a': '1'}])
+        table.put_item(Item={'p': p, 'c': c})
+        # TODO: change to 'MODIFY' when #6918 is fixed
+        events.append([['INSERT', 'MODIFY'], {'p': p, 'c': c}, {'p': p, 'c': c, 'a': '1'}, {'p': p, 'c': c}])
+        return events
+    do_test(test_table_sss_new_and_old_images_lsi, dynamodbstreams, do_updates, 'NEW_AND_OLD_IMAGES')
+
+def test_streams_putitem_new_item_overrides_old(test_table_ss_new_and_old_images, dynamodbstreams):
+    def do_updates(table, p, c):
+        events = []
+        table.put_item(Item={'p': p, 'c': c, 'a': 1})
+        # TODO: change to 'INSERT' when #6918 is fixed
+        events.append([['INSERT', 'MODIFY'], {'p': p, 'c': c}, None, {'p': p, 'c': c, 'a': 1}])
+        table.put_item(Item={'p': p, 'c': c})
+        # TODO: change to 'MODIFY' when #6918 is fixed
+        events.append([['INSERT', 'MODIFY'], {'p': p, 'c': c}, {'p': p, 'c': c, 'a': 1}, {'p': p, 'c': c}])
+        return events
+    do_test(test_table_ss_new_and_old_images, dynamodbstreams, do_updates, 'NEW_AND_OLD_IMAGES')
+
+def test_streams_putitem_new_items_override_old(test_table_ss_new_and_old_images, dynamodbstreams):
+    def do_updates(table, p, c):
+        events = []
+        table.put_item(Item={'p': p, 'c': c, 'a': 1})
+        # TODO: change to 'INSERT' when #6918 is fixed
+        events.append([['INSERT', 'MODIFY'], {'p': p, 'c': c}, None, {'p': p, 'c': c, 'a': 1}])
+        table.put_item(Item={'p': p, 'c': c, 'b': 2})
+        # TODO: change to 'MODIFY' when #6918 is fixed
+        events.append([['INSERT', 'MODIFY'], {'p': p, 'c': c}, {'p': p, 'c': c, 'a': 1}, {'p': p, 'c': c, 'b': 2}])
+        table.put_item(Item={'p': p, 'c': c, 'a': 3, 'b': 4})
+        # TODO: change to 'MODIFY' when #6918 is fixed
+        events.append([['INSERT', 'MODIFY'], {'p': p, 'c': c}, {'p': p, 'c': c, 'b': 2}, {'p': p, 'c': c, 'a': 3, 'b': 4}])
+        return events
+    do_test(test_table_ss_new_and_old_images, dynamodbstreams, do_updates, 'NEW_AND_OLD_IMAGES')
```

Comment:
Nitpick: do we actually need both this test and the one before it? Doesn't this test cover the previous one (or can cover it)? These streams tests are pretty slow, so having fewer of them is a benefit (of course only when the test isn't helpful).